crates/relayer/src/worker.rs

```diff
@@ -209,16 +210,24 @@ pub fn spawn_worker_tasks<ChainA: ChainHandle, ChainB: ChainHandle>(
         }
 
         Object::CrossChainQuery(cross_chain_query) => {
-            let (cmd_tx, cmd_rx) = crossbeam_channel::unbounded();
-            let cross_chain_query_task = cross_chain_query::spawn_cross_chain_query_worker(
-                chains.a.clone(),
-                chains.b,
-                cmd_rx,
-                cross_chain_query.clone(),
-            );
-            task_handles.push(cross_chain_query_task);
-
-            (Some(cmd_tx), None)
+            if config.chains.iter().any(|chain| match chain {
+                ChainConfig::CosmosSdk(c) => {
+                    (c.id == cross_chain_query.dst_chain_id) && c.allow_ccq
+                }
+            }) {
```

Comment:
Can we add an `allow_ccq` method on `ChainConfig` to simplify the code a bit?

```suggestion
            if config.chains.iter().any(|chain| chain.id() == cross_chain_query.dst_chain_id) && chain.allow_ccq()) {
```

Reply:
Done in 9d67e17f486652bcfb07609c74ed635fdb966473